Bipedal/bipedal_planner/bipedal_planner.py

```diff
@@ -0,0 +1,158 @@
+            vx_, vy_ = list(np.dot(rotate_mat_next, np.array([(1 + C) / (T_c * S) * x_, (C - 1) / (T_c * S) * y_])))
+            self.ref_p.append([px, py, f_theta])
+
+            # calculate reference COM
+            xd, xd_dot = px + x_, vx_
+            yd, yd_dot = py + y_, vy_
+
+            # calculate modified footsteps
+            D = a * math.pow(C - 1, 2) + b * math.pow(S / T_c, 2)
+            px_star = -a * (C - 1) / D * (xd - C * xi - T_c * S * xi_dot) - b * S / (T_c * D) * (xd_dot - S / T_c * xi - C * xi_dot)
+            py_star = -a * (C - 1) / D * (yd - C * yi - T_c * S * yi_dot) - b * S / (T_c * D) * (yd_dot - S / T_c * yi - C * yi_dot)
+            self.act_p.append([px_star, py_star, f_theta])
+
+            # plot
+            if plot:
+                # for plot trajectory, plot in for loop
+                for c in range(len(self.com_trajectory)):
+                    if c > len(com_trajectory_for_plot):
+                        # set up plotter
+                        plt.cla()
+                        ax.set_zlim(0, z_c * 2)                
+                        ax.set_aspect('equal', 'datalim')
+
+                        # update com_trajectory_for_plot
+                        com_trajectory_for_plot.append(self.com_trajectory[c])
+
+                        # plot com
+                        ax.plot([p[0] for p in com_trajectory_for_plot], [p[1] for p in com_trajectory_for_plot], [0 for p in com_trajectory_for_plot], color="red")
+
+                        # plot inverted pendulum
+                        ax.plot([px_star, com_trajectory_for_plot[-1][0]],
+                                [py_star , com_trajectory_for_plot[-1][1]],
+                                [0, z_c], color="green", linewidth=3)
+                        ax.scatter([com_trajectory_for_plot[-1][0]],
+                                   [com_trajectory_for_plot[-1][1]],
+                                    [z_c], color="green", s=300)
+                                   
+                        # foot rectangle for self.ref_p
+                        foot_width = 0.06
+                        foot_height = 0.04
+                        for i in range(len(self.ref_p)):
+                            angle = self.ref_p[i][2] + math.atan2(foot_height, foot_width) - math.pi
+                            r = math.sqrt(math.pow(foot_width / 3., 2) + math.pow(foot_height / 2., 2))
+                            rec = pat.Rectangle(xy = (self.ref_p[i][0] + r * math.cos(angle), self.ref_p[i][1] + r * math.sin(angle)),
+                                        width=foot_width, height=foot_height, angle=self.ref_p[i][2] * 180 / math.pi, color="blue", fill=False, ls=":")
+                            ax.add_patch(rec)
+                            art3d.pathpatch_2d_to_3d(rec, z=0, zdir="z")
+
+                        # foot rectangle for self.act_p                            
+                        for i in range(len(self.act_p)):
```

Comment:
Should change from i to j, because you used variable i before.

see:
https://lgtm.com/projects/g/AtsushiSakai/PythonRobotics/snapshot/723ee2368a5689bb10934d75b2befc827a807608/files/Bipedal/bipedal_planner/bipedal_planner.py?sort=name&dir=ASC&mode=heatmap#xefbb9f5c15ad194:1

Reply:
Oops, I fixed it.